tikv/raftstore/fsm_apply.go

```diff
@@ -136,6 +206,98 @@ type notifier struct {
 	router *router
 }
 
+type ApplyContext struct {
+	tag              string
+	timer            *util.SlowTimer
+	host             *CoprocessorHost
+	importer         *_import.SSTImporter
+	router           ApplyRouter
+	notifier         notifier
+	engines          Engines
+	cbs              util.MustConsumerVec
+	applyTaskResList []ApplyTaskRes
+	execCtx          *ExecContext
+	wb               *WriteBatch
+	wbLastBytes      uint64
+	wbLastKeys       uint64
+	lastAppliedIndex uint64
+	committedCount   uint64
+
+	enableSyncLog  bool
+	syncLogHint    bool
+	useDeleteRange bool
+}
+
+type WaitSourceMergeState struct {
+	pendingEntries []eraftpb.Entry
+	pendingMsgs    []Msg
+	readyToMerge   *atomic.Uint64
+	catchUpLogs    *CatchUpLogs
+}
+
+type ApplyDelegate struct {
+	id                uint64
+	term              uint64
+	region            metapb.Region
+	tag               string
+	stopped           bool
+	pendingRemove     bool
+	pendingCmds       PendingCmdQueue
+	merged            bool
+	isMerging         bool
+	lastMergeVersion  uint64
+	waitMergeState    *WaitSourceMergeState
+	readySourceRegion uint64
+	applyState        rspb.RaftApplyState
+	appliedIndexTerm  uint64
+	metrics           ApplyMetrics
+}
+
+type Destroy struct {
+	regionId uint64
+	peerId   uint64
+}
+
+type DelegateMailBox mailbox
+
+type CatchUpLogs struct {
+	targetMailBox DelegateMailBox
+	merge         raft_cmdpb.CommitMergeRequest
+	readyToMerge  *atomic.Uint64
+}
+
+type TaskResType int
+
+const (
+	ApplyTaskResType TaskResType = iota
+	DestroyResType
+)
+
+type TaskRes struct {
+	tp      TaskResType
+	destroy Destroy
+}
+
+type Builder struct {
+	tag             string
+	cfg             *Config
+	coprocessorHost *CoprocessorHost
+	importer        *_import.SSTImporter
+	engines         Engines
+	sender          notifier
+	router          ApplyRouter
+}
+
+type ApplyBatchSystem batchSystem
```

Comment:
If you define another type, the original types' method will be lost.

Reply:
Ok, thanks.